tdp/core/models/service_component_log.py

```diff
@@ -12,13 +12,19 @@
 class ServiceComponentLog(Base):
     __tablename__ = "service_component_log"
 
-    deployment_id = Column(Integer, ForeignKey("deployment_log.id"), primary_key=True)
-    service = Column(
-        String(length=SERVICE_NAME_MAX_LENGTH), primary_key=True, nullable=False
-    )
-    component = Column(
-        String(length=COMPONENT_NAME_MAX_LENGTH), primary_key=True, nullable=True
-    )
+    id = Column(Integer, primary_key=True)
+    deployment_id = Column(Integer, ForeignKey("deployment_log.id"), nullable=False)
+    service = Column(String(length=SERVICE_NAME_MAX_LENGTH), nullable=False)
+    component = Column(String(length=COMPONENT_NAME_MAX_LENGTH), nullable=True)
     version = Column(String(length=VERSION_MAX_LENGTH), nullable=False)
 
     deployment = relationship("DeploymentLog", back_populates="service_components")
+
+    __table_args__ = (
+        UniqueConstraint(
+            "deployment_id",
+            "service",
+            "component",
+            name="_unique_service_component_constraint",
```

Comment:
I think it is better to remove the first '_' character for ```_unique_service_component_constraint``` to avoid unexpected situation.
It seems that the name is not used by SQLite for the index. But it is used by PostgreSQL as name for the index of that constraint:
```
Indexes:
    "service_component_log_pkey" PRIMARY KEY, btree (id)
    "_unique_service_component_constraint" UNIQUE CONSTRAINT, btree (deployment_id, service, component)
```

If the name is not specified, the default Postgres index name for the constraint will be: 
```
<table>_<column1>_<column2>_<column3>_<...>_key
```
For our example, it will be:
```
service_component_log_deployment_id_service_component_key
```

Reply:
I removed the name to let the SGBD setup its own name